src/Database/Connection.php

```diff
@@ -278,16 +278,6 @@ public function run(Query $query): StatementInterface
         return $this->getDisconnectRetry()->run(fn () => $this->_driver->run($query));
     }
 
-    /**
-     * Create a new Query instance for this connection.
-     *
-     * @return \Cake\Database\Query
-     */
-    public function newQuery(): Query
-    {
-        return new Query($this);
-    }
```

Comment:
I wonder how big if a break this will be. There's nothing we could transform calls to newQuery() into.

Reply:
We *could* backport the `newSelectQuery()` methods to 4.x and add a deprecation to `newQuery()`. That would let us encourage users upgrade in a reasonable timeframe.

I'm happy to take care of that when this merges.